HexDump/HexDump_Parse.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Drew Noakes. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace HexDump
+{
+    public  static partial class HexDump
+    {
+        
+        private static readonly Regex _re =
+            new Regex(@"^((?<offset>[0-9a-f]+)\s+)?(?<hexa>[0-9a-f\s]{48})\s+(?<dump>.+)?$",
+                RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        /// <summary>
+        /// Parse HexDump into a byte array
+        /// Support only hexdumped bytes array with these attributes:
+        /// - columnWidth = 8
+        /// - columnCount = 2
+        /// - includeOffset = true
+        /// - includeAscii = true
+        ///
+        /// </summary>
+        /// <param name="dump"></param>
+        /// <returns></returns>
+        public static byte[] Parse(string dump )
+        {
+            //00000000   01 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00   ................
+            //00000000   01 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00   ................
+
+            var result = new List<byte>();
+            var lines = dump.Split(Environment.NewLine.ToCharArray());
```

Comment:
Using `Environment.NewLine` means you can get different behaviours on different machines for the same input. This kind of thing regularly breaks CI, for example. I'd rather see split explicitly on `'\r'` and `'\n'` and remove empty entries (though I'd rather not use `String.Split` at all as it allocates an array and temporary strings, both of which can be avoided).

Reply:
Changed it to use StringReader and remove the ToArray() in linq query.